pallets/baby-liminal/src/lib.rs

```diff
@@ -222,12 +275,19 @@ pub mod pallet {
                 Error::<T>::IdentifierAlreadyInUse
             );
 
+            let deposit = T::VerificationKeyDepositAmount::get();
+            T::Currency::reserve(&who, deposit).map_err(|_| Error::<T>::CannotAffordDeposit)?;
+
             VerificationKeys::<T>::insert(
                 identifier,
                 BoundedVec::try_from(key).unwrap(), // must succeed since we've just check length
             );
 
-            Self::deposit_event(Event::VerificationKeyStored);
+            // will never overwrite anything since we already check the VerificationKeys map
+            VerificationKeyOwners::<T>::insert(identifier, &who);
+            VerificationKeyDeposits::<T>::insert(&who, deposit);
```

Comment:
this will overwrite the previous deposit, won't it?

Reply:
it's under a tuple now